jax/_src/pallas/triton/lowering.py

```diff
@@ -1620,9 +1620,9 @@ def _broadcast_in_dim_lowering_rule(
 
 
 @register_lowering(lax.squeeze_p)
-def _squeeze_lowering_rule(ctx: LoweringRuleContext, a, *, dimensions):
+def _squeeze_lowering_rule(ctx: LoweringRuleContext, a, *, dimensions, sharding=None):
```

Comment:
Do you have a reproducer for this? AFAICT the MLIR lowering rule for `squeeze_p` doesn't expect the `sharding` argument: https://github.com/jax-ml/jax/blob/d0f63da4b530950caeb9425429c42c1a9ac3c76a/jax/_src/lax/lax.py#L4650.

Reply:
I see. Then does it make sense to just modify the `_reshape_lowering_rule` to be
  `return _reshape_lowering_rule(ctx, a, new_sizes=None, dimensions=None, sharding=None) `?
  
  The main reason is that this PR (https://github.com/jax-ml/jax/pull/25103/) will lead to error if there is no sharding arg passed to `_reshape_lowering_rule`

Reply:
I might, but given that Pallas is released with JAX, it seems unlikely that there will ever be a `reshape_p.bind` call which doesn't pass `sharding` at all. Apologies if I'm missing something, a reproducer I can run will clear all of this up.